src/stream/src/executor/managed_state/join/join_entry_state.rs

```diff
@@ -32,205 +22,54 @@ type JoinEntryStateValues<'a> = btree_map::Values<'a, PkType, StateValueType>;
 
 type JoinEntryStateValuesMut<'a> = btree_map::ValuesMut<'a, PkType, StateValueType>;
 
-/// We manages a `BTreeMap` in memory for all entries belonging to a join key,
-/// since each `WriteBatch` is an ordered list of key-value pairs.
-/// When evicted, `BTreeMap` does not hold any entries.
-pub struct JoinEntryState<S: StateStore> {
+/// We manages a `BTreeMap` in memory for all entries belonging to a join key.
+/// When evicted, `cached` does not hold any entries.
+/// If a `JoinEntryState` exists for a join key, the all records under this
+/// join key will be presented in the cache.
+pub struct JoinEntryState {
```

Comment:
Seems this struct is definitely a `BTreeMap` now. 🤣 How about implement `Deref` for it?